### DISCUSSION TITLE
How to delete commands.

### DISCUSSION THREAD

**Discussion by EdsonMatt** [eternnoir/pyTelegramBotAPI#1714] [Q&A]
Hello guys! 

I took a look at how to set commands in the examples available, in my case AsyncTelebot. Well, the procedure to set the commands seem to work fine, although I'm getting this warning:

![image](https://user-images.githubusercontent.com/111212344/189534796-94e3eb43-2861-4334-89ba-0cd111d6bdc2.png)

On the other hand, using the procedure to delete the commands delete them at first. If I leave the private chat with the bot and come back (in order to update, like described in the example) the menu disappear, which is fine and intended. But if I leave the chat again and come back, the commands are shown again, which is not intended.

![image](https://user-images.githubusercontent.com/111212344/189534709-86ed04b0-0706-4c2e-b910-3110b58a72b7.png)

I don't know if this has something to do with the warning I'm getting about “Unclosed client session” and “Unclosed connector”, but this doesn't happen with the Telebot, just with the Async one. Any suggestion?

(I'm using the same code in the example).

**Comment by coder2020official** [comment]
The example needs to be updated by calling a function to close session.
So, you are running [this example?](https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/asynchronous_telebot/set_command_example.py)
If so - this deletes the commands and sets the commands. So you should get those commands.

**Reply by EdsonMatt** [reply]
Yes, this is the example I'm running.

What you said make sense, but I'm still facing the problem in the following scenario:

If I run the example excluding the line:

`await bot.delete_my_commands(scope=None, language_code=None)`

The commands are set, like it would supposed to be. But, if I run the example just with the above line, without the set part, the commands are deleted at first, but If I leave the private chat with the bot and come back, the menu command is there again.

**Reply by coder2020official** [reply]
I hope you are sure that you are not resetting the commands again. Quit telegram, and reopen it.

**Reply by EdsonMatt** [reply]
What do you mean by resetting the commands again? Sorry I'm kind of new to these stuff.

But I think I'm not resetting the commands. If I run the following code in order to delete the commands, I still can see the menu commands, even after quitting and reopening Telegram.

```
import asyncio
from telebot.async_telebot import AsyncTeleBot

API_TOKEN = 'token here'
bot = AsyncTeleBot(API_TOKEN)


async def main():

    # use in for delete with the necessary scope and language_code if necessary
    await bot.delete_my_commands(scope=None, language_code=None)

if __name__ == '__main__':
    asyncio.run(main())
```

![image](https://user-images.githubusercontent.com/111212344/189551903-e5c23034-9345-4860-8a58-e3306fc550ce.png)


By the way, running this I get the same warning about 'Unclosed client session' and 'Unclosed connector'.
